### DISCUSSION TITLE
Something exploded for no reason.

### DISCUSSION THREAD

**Discussion by pavel4win** [FakeFishGames/Barotrauma#14529] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

Docked to the outpost
I spent 2 hours cleaning the boat
When my bot was near the cabinet connected to the fabricator, something suddenly exploded and then a fire broke out.
The bot died almost instantly and I had to restart the round.
The bot did not have any hazardous materials at that moment
There were uex and c-4 in the cabinet
I couldn't even figure out what exactly happened, what exploded?
Is there really a chance that the explosives will detonate for no reason?
I think this is too much.
What If I will store explosives in a storage container, will this prevent the explosives from detonating for no reason?

### Reproduction steps

I do not know how to replicate

### Bug prevalence

Just once

### Single player or multiplayer?

Single player

### Version

Other

### -

1.5.9.1

### Which operating system did you encounter this bug on?

Windows

**Comment by kaedepon** [comment]
- Did `Volatile Compound N` deteriorate completely?
- Sodium, Lithium and Potassium explode when mixed with water, is there a different reason?

This is just FYI the slots in Fabricator and Deconstructor are not safely stored.

**Reply by pavel4win** [reply]
I know all of the above.
There was no water in the compartment.
Sodium, Lithium were in the cabinet. But the cabinet is waterproof.
I never craft Volatile Compound N. And I never bring it from sunken boats.
At the time of the explosion, the boat had already been at the outpost for 2 hours.
The explosion occurred without any reason.
The cabinets are not fireproof.
Once, during a fire, I noticed spontaneous  detonation IC-4 lying in the cabinet.
But yesterday the fire occurred after the explosion, not before it.

My only guess is spontaneous detonation of UEX or C-4
Is this possible?
Another option is that the explosion occurred in the fabricator during crafting. 
Is this possible?
But I can't remember if anything was being crafted at the time of the explosion.

**Reply by kaedepon** [reply]
> My only guess is spontaneous detonation of UEX or C-4
> Another option is that the explosion occurred in the fabricator during crafting.

Hmm.. I think these can be not possible.
There must be some other reason, but I can't think of any :(

**Reply by SAYtan47** [reply]
I actually had the same strange explosion but in different conditions. I took my mechanic bot out of the sub to mine minerals, he swam for like 5 seconds and exploded for no reason. The explosion in range was about 3 meters and it killed him almost instantly. At first i thought there were some chemicals with him at the moment, but when I restarted the level there was nothing on bot that could explode and I definetly remember not giving him anything explosive in the mission. He just had a flashy heart attack.

**Comment by Ek-Videogames** [comment]
Just a hunch, but this sorta sounds like a possible 'condition' bug, maybe related to fire or itemdamage?

Was there anything like air tanks or fuel tanks around? Was any explosives being used in crafting at all?

( For those who don't know, its possible & there's been bugs in the past related to fuel tanks or oxygen tanks detonating because of being 'broken', or non-volatile explosives like UEX/C-4 detonating when being consumed by a crafting recipe because of how the game was counting them as 'destroyed' )

Or maybe some changes to item tagging logic or AI 'item search' behavior has caused a bug where bots are suddenly loading the wrong fuel tanks into welders and then blowing up randomly at a later moment.
